dev/com.ibm.ws.microprofile.config/src/com/ibm/ws/microprofile/config/archaius/composite/PollingDynamicConfig.java

```diff
@@ -270,4 +253,47 @@ public String getSourceID() {
         return this.id;
     }
 
+    /**
+     * Runnable which calls {@link PollingDynamicConfig#update()}
+     * <p>
+     * This class only holds a weak reference to the PollingDynamicConfig to ensure that it won't keep it alive after the ConfigImpl that uses it has been garbage collected
+     */
+    private static class Refresher implements Runnable {
+        private final WeakReference<PollingDynamicConfig> configRef;
+        private volatile Future<?> future;
+
+        private Refresher(PollingDynamicConfig config) {
+            configRef = new WeakReference<>(config);
+        }
+
+        @Override
+        public void run() {
+            PollingDynamicConfig config = configRef.get();
+            if (config == null) {
+                // Our pollingDynamicConfig has been GC'd, we can't update it any more, cancel ourselves
+                if (future != null) {
+                    future.cancel(false);
```

Comment:
instead of calling cancel directly, suggest this should call the existing close method. That close method could do with some polishing too!

Reply:
We can't call the `close()` method.

This code is part of the recurring scheduled task and this bit is hit when the `PollingDynamicConfig` instance has been GC'd, so there's no instance to call close on.

Reply:
@tevans78 I've merged it without making any change here. If we can come up with an improvement here, we can do it in another PR